### PR TITLE
80 epic annotation target

### PR DESCRIPTION
**Pull request by Drilio**
Merge Wip-Surface-Transformer into mui5-tetras-main-stable and lint files. Bug on arrow reported #14

### CODE REVIEW

File: src/annotationForm/AnnotationFormFooter.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ function AnnotationFormFooter({
    */
   const submitAnnotationForm = async (e) => {
     e.preventDefault();
+    return console.log(state);
```
[the remark this answers is not in the document]
Comment (Drilio) [reply src/annotationForm/AnnotationFormFooter.js]: done

File: src/annotationForm/AnnotationFormOverlay/KonvaDrawing/Surface.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}) {
+  const shapeRef = useRef();
+  const trRef = useRef();
+
+  useEffect(() => {
+    if (trRef.current) {
+      trRef.current.nodes([shapeRef.current]);
+      trRef.current.getLayer().batchDraw();
+    }
+  }, [tabView]);
```
Comment (geourjoa) [inline src/annotationForm/AnnotationFormOverlay/KonvaDrawing/Surface.js:21]: Remove useless useEffect hook

**Comment by geourjoa** [comment]
The image tool in the overview view is not working now

File: src/annotationForm/AnnotationFormOverlay/KonvaDrawing/shapes/Freehand.js [hunk 3]
```diff
@@ -15,19 +15,20 @@ function Freehand({
       trRef.current.nodes([shapeRef.current]);
       trRef.current.getLayer().batchDraw();
     }
+    console.log('SHAPE', shape);
```
Comment (geourjoa) [inline src/annotationForm/AnnotationFormOverlay/KonvaDrawing/shapes/Freehand.js:18]: Remove useless console.log

File: src/annotationForm/AnnotationFormOverlay/KonvaDrawing/shapes/ArrowNode.js [hunk 4]
```diff
@@ -37,10 +45,11 @@ function ArrowNode({
         id={shape.id}
         draggable={activeTool === 'cursor' || activeTool === 'edit'}
         onClick={handleClick}
+        onMousedown={handleClick}
```
Comment (geourjoa) [inline src/annotationForm/AnnotationFormOverlay/KonvaDrawing/shapes/ArrowNode.js:48]: Why did you add an other listener ? (like all other kanva component)

File: src/annotationForm/AnnotationFormTarget.js [hunk 5]
```diff
@@ -113,28 +112,30 @@ function AnnotationFormTarget({
   };
 
   /** Change from Tstart HMS Input */
-  const updateTstart = (value) => {
-    if (value > tend) {
+  const updateTstart = (valueTstart) => {
```
Comment (geourjoa) [inline src/annotationForm/AnnotationFormTarget.js:115]: Rename valueTStart in newValueTStart

File: src/annotationForm/AnnotationDrawing.js [hunk 6]
```diff
@@ -1,26 +1,43 @@
...
+  const [surfacedata, setSurfaceData] = useState({
+    height: height / props.scale,
+    scaleX: 1,
+    scaleY: 1,
+    width: width / props.scale,
+    x: 1,
+    y: 1,
+  });
 
   useEffect(() => {
```
Comment (geourjoa) [inline src/annotationForm/AnnotationDrawing.js:28]: Useless useEffect

File: src/annotationForm/AnnotationDrawing.js [hunk 7]
```diff
@@ -43,22 +60,100 @@ function AnnotationDrawing({ drawingState, setDrawingState, height, width, ...pr
       currentShape: shape,
       shapes: [...drawingState.shapes, shape],
     });
-  }, [props.imageEvent]);
+  }, [drawingState, imageEvent, props.imageEvent, setDrawingState]);
```
Comment (geourjoa) [inline src/annotationForm/AnnotationDrawing.js:63]: Why did you add these new dependancy in useEffect ?
UseEffect can be removed I thing (need to check)

File: src/annotationForm/AnnotationDrawing.js [hunk 8]
```diff
@@ -43,22 +60,100 @@ function AnnotationDrawing({ drawingState, setDrawingState, height, width, ...pr
...
+      setDrawingState({
+        ...drawingState,
+        currentShape,
+        shapes: [...drawingState.shapes, currentShape],
+      });
+    }
+  };
+
+  /** */
+  useEffect(() => {
```
Comment (geourjoa) [inline src/annotationForm/AnnotationDrawing.js:89]: Remove this useEffect

**Comment by geourjoa** [comment]
The app crash when opening a image manifest
